Sources/Common/DataModel/PolyLine/index.js

```diff
@@ -66,13 +66,77 @@ function vtkPolyLine(publicAPI, model) {
 
     return line.evaluateLocation(pcoords, x, weights);
   };
+
+  publicAPI.evaluateOrientation = (subId, pcoords, q, weights) => {
+    if (model.orientations) {
+      line.setOrientations([
+        model.orientations[subId],
+        model.orientations[subId + 1],
```

Comment:
what if subId is the last from the polyline ?

Reply:
`subId` can't be the last point of the polyline

`evaluateOrientation` interpolates the orientation between two points
The subId has to be in `[0, nPoints - 1[` (in `[0, nSegments[`)
This is part of the contract of the function, I can make it clear in the ts header if you want

Also `findPointIdAtDistanceFromFirstPoint` will never return `nPoints - 1` because it is meant to be used with this function
This is why it used to be called `findSubId`

Reply:
fair enough. leave it as is.